ctags/parsers/c.c

```diff
@@ -1221,11 +1219,10 @@ static void addOtherFields (tagEntryInfo* const tag, const tagType type,
 			{
 				tag->extensionFields.access = accessField (st);
 			}
-            if ((true == st->gotArgs) && (true == Option.extensionFields.argList) &&
```

Comment:
Just checked it an it's handled by renderExtensionFieldMaybe() inside main where isFieldEnabled(FIELD_SIGNATURE) is being checked. So there's no need for the check in the parser.